include/deal.II/multigrid/mg_coarse.h

```diff
@@ -329,6 +339,22 @@ MGCoarseGridApplySmoother<VectorType>::initialize (const MGSmootherBase<VectorTy
     (&coarse_smooth_,typeid(*this).name());
 }
 
+
+template<class VectorType>
+void
+MGCoarseGridApplySmoother<VectorType>::clear()
+{
+  coarse_smooth = nullptr;
+}
+
+
+template<class VectorType>
+MGCoarseGridApplySmoother<VectorType>::~MGCoarseGridApplySmoother()
+{
+  clear();
```

Comment:
My point was that calling `clear` does only unsubscribe from the respective variable, which is the same as the destructor of `SmartPointer` does already. So you can remove it from `MGCOarseGridLACIteration` as well I guess...